doc/src/build/programming-with-objects/ch3-immutable-objects.md

```diff
@@ -0,0 +1,150 @@
+
+### Create immutable object
+
+Regardless of whether an object was just created or already owned by an account, to turn this object into an immutable object, we need to call the following API in the [Transfer Library](../../../../sui_programmability/framework/sources/Transfer.move):
+```rust
+public native fun freeze_object<T: key>(obj: T);
+```
+After this call, the specified object will become permanently immutable. This is a non-reversible operation; hence, freeze an object only when you are certain that it will never need to be mutated.
+
+Let's add an entry function to the [ColorObject](../../../../sui_programmability/examples/objects_tutorial/sources/ColorObject.move) module to turn an existing (owned) `ColorObject` into an immutable object:
+```rust
+public(script) fun freeze_object(object: ColorObject, _ctx: &mut TxContext) {
+    Transfer::freeze_object(object)
+}
+```
+In the above function, one must already own a `ColorObject` to be able to pass it in. At the end of this call, this object is *frozen* and can never be mutated. It is also no longer owned by anyone.
+> :bulb: Note the `Transfer::freeze_object` API requires passing the object by value. Had we allowed passing the object by a mutable reference, we would then still be able to mutate the object after the `freeze_object` call; this contradicts the fact that it should have become immutable.
+
+Alternatively, you can also provide an API that creates an immutable object at birth:
+```rust
+public(script) fun create_immutable(red: u8, green: u8, blue: u8, ctx: &mut TxContext) {
+    let color_object = new(red, green, blue, ctx);
+    Transfer::freeze_object(color_object)
+}
+```
+In this function, a fresh new `ColorObject` is created and immediately turned into an immutable object before being owned by anyone.
+
+### Use immutable object
+Once an object becomes immutable, the rules of who could use this object in Move calls change:
+1. An immutable object can be passed only as a read-only reference to Move entry functions as `&T`.
+2. Anyone can use immutable objects. Immutable objects are considered shared.
+
+Recall that we defined a function that copies the value of one object to another:
+```rust
+public(script) fun copy_into(from_object: &ColorObject, into_object: &mut ColorObject, _ctx: &mut TxContext);
+```
+In this function, anyone can pass an immutable object as the first argument `from_object`, but not the second argument.
+
+Since immutable objects can never be mutated, there will never be a data race even when multiple transactions are using the same immutable object at the same time. Hence, the existence of shared immutable objects does not pose any requirement on consensus.
```

Comment:
missed this, but it can be fixed later (if it needs fixing)

This section refers to them as "shared" but I thought we were trying to avoid that usage?

Reply:
ah good catch